Main/Source/actions.cpp

```diff
@@ -161,7 +161,15 @@ void consume::Terminate(truth Finished)
 
   if(Finished)
   {
-    if(Consuming->Exists() && !game::IsInWilderness() && (!Actor->IsPlayer() || ivanconfig::GetAutoDropLeftOvers()))
+    truth PlayerWantsToDiscard = ivanconfig::GetAutoDropLeftOvers();
+    if(Consuming->GetSecondaryMaterial()){
+      PlayerWantsToDiscard = (
+        ivanconfig::GetAutoDropLeftOvers() &&
+        (Consuming->GetSecondaryMaterial()->GetVolume() == 0) // don't drop after tasting
+      );
+    }
+
+    if(Consuming->Exists() && !game::IsInWilderness() && (!Actor->IsPlayer() || PlayerWantsToDiscard))
```

Comment:
`Consuming->Exists()` seems unnecessary because if `game::SearchItem()` succeeds the item's memory is not yet freed. Should check `Consuming != NULL` instead.

`entity::Exists()` is a really confusing function.

Reply:
Need to check both: `Consuming && Consuming->Exists()`, before calling other member functions.

Reply:
@iology good call, I wrapped everything inside a `Consuming && Consuming->Exsists()`